Sming/system/esp8266_phy.c

```diff
@@ -0,0 +1,268 @@
+
+    // pwr_ind_11b_0
+    // 1m, 2m power index [0~5]
+    [97] = 0,
+
+    // pwr_ind_11b_1
+    // 5.5m, 11m power index [0~5]
+    [98] = 0,
+
+    // vdd33_const
+    // the voltage of PA_VDD
+    // x=0xff: it can measure VDD33,
+    // 18<=x<=36: use input voltage,
+    // the value is voltage*10, 33 is 3.3V, 30 is 3.0V,
+    // x<18 or x>36: default voltage is 3.3V
+    //
+    // the value of this byte depend from the TOUT pin usage (1 or 2):
+    // 1)
+    // analogRead function (system_adc_read()):
+    // is only available when wire TOUT pin17 to external circuitry, Input Voltage Range restricted to 0 ~ 1.0V.
+    // For this function the vdd33_const must be set as real power voltage of VDD3P3 pin 3 and 4
+    // The range of operating voltage of ESP8266 is 1.8V~3.6V，the unit of vdd33_const is 0.1V，so effective value range of vdd33_const is [18,36]
+    // 2)
+    // getVcc function (system_get_vdd33):
+    // is only available when TOUT pin17 is suspended (floating), this function measure the power voltage of VDD3P3 pin 3 and 4
+    // For this function the vdd33_const must be set to 255 (0xFF).
+    [107] = 33,
+
+    // disable RF calibration for certain number of times
+    [108] = 0,
+
+    // freq_correct_en
+    // bit[0]:0->do not correct frequency offset, 1->correct frequency offset.
+    // bit[1]:0->bbpll is 168M, it can correct + and - frequency offset,  1->bbpll is 160M, it only can correct + frequency offset
+    // bit[2]:0->auto measure frequency offset and correct it, 1->use 113 byte force_freq_offset to correct frequency offset.
+    // 0: do not correct frequency offset.
+    // 1: auto measure frequency offset and correct it,  bbpll is 168M, it can correct + and - frequency offset.
+    // 3: auto measure frequency offset and correct it,  bbpll is 160M, it only can correct + frequency offset.
+    // 5: use 113 byte force_freq_offset to correct frequency offset, bbpll is 168M, it can correct + and - frequency offset.
+    // 7: use 113 byte force_freq_offset to correct frequency offset, bbpll is 160M , it only can correct + frequency offset.
+    [112] = 0,
+
+    // force_freq_offset
+    // signed, unit is 8kHz
+    [113] = 0,
+
+    // rf_cal_use_flash
+    // 0: RF init no RF CAL, using all RF CAL data in flash, it takes about 2ms for RF init
+    // 1: RF init only do TX power control CAL, others using RF CAL data in flash, it takes about 20ms for RF init
+    // 2: RF init no RF CAL, using all RF CAL data in flash, it takes about 2ms for RF init  (same as 0?!)
+    // 3: RF init do all RF CAL, it takes about 200ms for RF init
+    [114] = 1
+};
+
+#ifndef F_CRYSTAL
+#define F_CRYSTAL 40000000
+#endif
+
+extern int __real_register_chipv6_phy(uint8_t* initData);
+
+int __attribute__((weak)) get_adc_mode()
+{
+	return 33;
+}
+
+int ICACHE_FLASH_ATTR __attribute__((weak)) custom_register_chipv6_phy(uint8_t* initData)
+{
+	if (initData != NULL) {
+	  memcpy(initData, phyInitData, sizeof(phyInitData));
```

Comment:
Shouldn't this be memcpy_P?

Reply:
I don't think that is needed but will test it later just to be sure.